meta-dts-distro/recipes-dts/dts-scripts/dts-scripts_git.bb

```diff
@@ -0,0 +1,44 @@
+SUMMARY = "Dasharo Tools Suite scripts."
+SECTION = "tools"
+HOMEPAGE = "https://github.com/Dasharo/dts-v2"
+
+LICENSE = "MIT"
+LIC_FILES_CHKSUM = "file://LICENSE;md5=575ab8837c52eafc7b9eb8b244072bc7"
+
+SRCREV = "13dad8ea6901cf921de0dd45d464bea3535892e0"
+SRC_URI = "git://github.com/Dasharo/dts-v2;protocol=https;branch=main"
+
+PV = "0.1+git${SRCPV}"
+
+S = "${WORKDIR}/git"
+
+RDEPENDS:${PN} = " \
+    packagegroup-coreboot-utils \
+    flashrom \
+    iasl \
+    tar \
+    i2c-tools \
+    i2c-tools-misc \
+    bash \
+    dmidecode \
+    acpi-call-dkms \
+    iotools \
+"
+
+do_install () {
+    install -d ${D}/${sbindir}
```

Comment:
You may consider adding Makefile to the dts-v2 repo, and simply call `oe_runmake install 'DESTDIR=${D}'` here. That would make it easier to install these script for testing as well, in some test environment.

Reply:
I suppose this Makefile will only have two rules: to install these scripts in `sbin` and to remove it from `sbin`?

Reply:
Only install would do as well. It should do the same as in the recipe, simply inside a Makefile, which makes it more self-contained (closer to the script), so the recipe does not need to change the installation methods when we change the file paths for instance (and I am sure we will do that).

Reply:
I do not have an example on hand, but I am sure you will find plenty. The one I recently changed was: https://github.com/Dasharo/osfv-scripts/blob/main/osfv_cli/Makefile#L10

The DESTDIR is a standard variable: https://www.gnu.org/prep/standards/html_node/DESTDIR.html

It can be defined in Makefile as some default, but with the `?=`, so it can be overriden in Yocto recipe

Reply:
I have been experimenting for a while with Makefile which I have added in f8afaca3873821d13a9db79c944302d7ff9df259 and it seems that it is not working as I expected with Yocto.

With following `do_install`:
```bb
do_install () {
    install -d ${D}${sbindir}
    install -d ${D}${sysconfdir}/profile.d
    oe_runmake install 'DESTDIR=${D}${sbindir} SYSCONFDIR=${D}${sysconfdir}/profile.d/'
}
```
I am getting following error during build:
```bash
ERROR: dts-scripts-0.1+gitAUTOINC+f8afaca387-r0 do_compile: oe_runmake failed
ERROR: dts-scripts-0.1+gitAUTOINC+f8afaca387-r0 do_compile: ExecutionError('/build/tmp/work/core2-64-dts-linux/dts-scripts/0.1+gitAUTOINC+f8afaca387-r0/temp/run.do_compile.229', 1, None, None)
ERROR: Logfile of failure stored in: /build/tmp/work/core2-64-dts-linux/dts-scripts/0.1+gitAUTOINC+f8afaca387-r0/temp/log.do_compile.229
Log data follows:
| DEBUG: Executing shell function do_compile
| NOTE: make -j 8
| install -D -m 0755 $(find ./ -type f -name dts-profile.sh) //dts-profile.sh
| install -D -m 0655 $(find ./ -type f -name dts-environment.sh) /dts-environment.sh
| install -D -m 0655 $(find ./ -type f -name dts-functions.sh) /dts-functions.sh
| install -D -m 0755 $(find ./ -type f -name cloud_list) /cloud_list
| install -D -m 0755 $(find ./ -type f -name dasharo-deploy) /dasharo-deploy
| install -D -m 0755 $(find ./ -type f -name dts) /dts
| install -D -m 0755 $(find ./ -type f -name dts-boot) /dts-boot
| install -D -m 0755 $(find ./ -type f -name ec_transition) /ec_transition
| install: cannot create regular file '//dts-profile.sh': Permission denied
| install: cannot create regular file '/dts-functions.sh'install: cannot create regular file '/cloud_list': Permission denied
| install: cannot create regular file '/dts-environment.sh': Permission denied
| : Permission denied
| make: *** [Makefile:14: //dts-profile.sh] Error 1
| make: *** Waiting for unfinished jobs....
| make: *** [Makefile:17: /dts-environment.sh] Error 1
| make: *** [Makefile:17: /dts-functions.sh] Error 1
| install: cannot create regular file '/dasharo-deploy'make: *** [Makefile:20: /cloud_list] Error 1
| : Permission denied
| install: cannot create regular file '/dts': Permission denied
| make: *** [Makefile:20: /dasharo-deploy] Error 1
| make: *** [Makefile:20: /dts] Error 1
| install: cannot create regular file '/dts-boot': Permission denied
| make: *** [Makefile:20: /dts-boot] Error 1
| install: cannot create regular file '/ec_transition': Permission denied
| make: *** [Makefile:20: /ec_transition] Error 1
| ERROR: oe_runmake failed
| WARNING: exit code 1 from a shell command.
ERROR: Task (/repo/meta-dts-distro/recipes-dts/dts-scripts/dts-scripts_git.bb:do_compile) failed with exit code '1'
NOTE: Tasks Summary: Attempted 4171 tasks of which 4170 didn't need to be rerun and 1 failed.

Summary: 1 task failed:
  /repo/meta-dts-distro/recipes-dts/dts-scripts/dts-scripts_git.bb:do_compile
Summary: There were 2 ERROR messages, returning a non-zero exit code.
2024-04-09 09:38:20 - ERROR    - Command returned non-zero exit status 1
```
I did not add any modifications to `do_compile`, so the warning seems strange.

After some digging I have found the answer: If you start using `oe_runmake` somewhere in a recipe - Yocto automatically adds `oe_runmake` to `do_config` and `do_compile` in `poky/meta/classes/base.bbclass`:
```bbclass
addtask configure after do_patch
do_configure[dirs] = "${B}"
base_do_configure() {
	if [ -n "${CONFIGURESTAMPFILE}" -a -e "${CONFIGURESTAMPFILE}" ]; then
		if [ "`cat ${CONFIGURESTAMPFILE}`" != "${BB_TASKHASH}" ]; then
			cd ${B}
			if [ "${CLEANBROKEN}" != "1" -a \( -e Makefile -o -e makefile -o -e GNUmakefile \) ]; then
				oe_runmake clean
			fi
			# -ignore_readdir_race does not work correctly with -delete;
			# use xargs to avoid spurious build failures
			find ${B} -ignore_readdir_race -name \*.la -type f -print0 | xargs -0 rm -f
		fi
	fi
	if [ -n "${CONFIGURESTAMPFILE}" ]; then
		mkdir -p `dirname ${CONFIGURESTAMPFILE}`
		echo ${BB_TASKHASH} > ${CONFIGURESTAMPFILE}
	fi
}

addtask compile after do_configure
do_compile[dirs] = "${B}"
base_do_compile() {
	if [ -e Makefile -o -e makefile -o -e GNUmakefile ]; then
		oe_runmake || die "make failed"
	else
		bbnote "nothing to compile"
	fi
}
```

Reply:
So, I think we should stay without `oe_runmake` in recipe but with Makefile in `Dasharo/dts-v2`.

Reply:
> After some digging I have found the answer: If you start using oe_runmake somewhere in a recipe - Yocto automatically adds oe_runmake to do_config and do_compile in poky/meta/classes/base.bbclass:

If this is the only blocker, simply use `do_configure[noexec] = "1"` for the stage you with not to execute. I have not looked into detailes of makefile yet, please let me know if I need to at this point.

Reply:
Thanks for the tip, add installing via Makefile here d10b32b1efbd182f43efc042501b99036eb7c6cd.